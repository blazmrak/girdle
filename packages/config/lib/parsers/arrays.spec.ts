import { initializeEnvironment } from '../index'
import { expectTypeOf } from 'expect-type'
import { Var } from './index'

describe('array', () => {
  describe('type', () => {
    it('expect error if type of result is not string array for array that should contain strings', () => {
      const env = { ARR: '1' }
      const config = initializeEnvironment(
        {
          arr: Var.array('string'),
        },
        { env },
      )

      // @ts-expect-error
      expectTypeOf(config).toEqualTypeOf<{
        arr: number[]
      }>()
    })
  })

  describe('of strings', () => {
    it('parses basic options', () => {
      const env = { ARR: 'str1, whitespace_around   ,3,' }

      const config = initializeEnvironment(
        {
          defaultArr: Var.array('string').default(['d1', 'd2']),
          arr: Var.array('string'),
        },
        { env },
      )

      expect(config).toMatchSnapshot()
      expectTypeOf(config).toEqualTypeOf<{
        defaultArr: string[]
        arr: string[]
      }>()
    })
  })

  describe('of numbers', () => {
    it('parses basic options', () => {
      const env = { ARR: '5,6' }

      const config = initializeEnvironment(
        {
          defaultArr: Var.array('number').default([1, 2]),
          arr: Var.array('number'),
          overrides: Var.array('number').default([3]).name('ARR'),
        },
        { env },
      )

      expect(config).toMatchSnapshot()
      expectTypeOf(config).toEqualTypeOf<{
        defaultArr: number[]
        arr: number[]
        overrides: number[]
      }>()
    })
  })
})
