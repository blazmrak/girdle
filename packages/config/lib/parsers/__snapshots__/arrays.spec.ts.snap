// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array parser of enums parses basic options 1`] = `
{
  "arr": [
    "test",
    "is",
  ],
  "arrEnum": [
    "test",
    "is",
  ],
  "defaultArr": [
    "test",
  ],
  "env": "local",
  "overrides": [
    "test",
    "is",
  ],
}
`;

exports[`array parser of numbers parses basic options 1`] = `
{
  "arr": [
    5,
    6,
  ],
  "defaultArr": [
    1,
    2,
  ],
  "env": "local",
  "overrides": [
    5,
    6,
  ],
}
`;

exports[`array parser of strings parses basic options 1`] = `
{
  "arr": [
    "str1",
    "whitespace_around",
    "3",
  ],
  "defaultArr": [
    "d1",
    "d2",
  ],
  "env": "local",
}
`;
